lightning/src/routing/scoring.rs

```diff
@@ -135,6 +137,32 @@ pub trait LockableScore<'a> {
 	fn lock(&'a self) -> Self::Locked;
 }
 
+/// Refers to a scorer that is accessible under lock and also writeable to disk
+///
+/// We need this trait to be able to pass in a scorer to `lightning-background-processor` that will enable us to
+/// use the Persister to persist it.
+pub trait WriteableScore<'a>: LockableScore<'a> + Writeable {}
+
+impl<'a, T> WriteableScore<'a> for T
+	where T: LockableScore<'a> + Writeable
+{}
+
+impl<'a, T> LockableScore<'a> for Arc<T>
```

Comment:
This can be blanket, no? Instead of implementing it only for `Arc` we can implement it for any `Deref<T> where T: LockableScore and T::Locked: Writeable`.

Reply:
touché, not sure why I didn't think of that

Reply:
@TheBlueMatt I tried implementing a blanket using what you suggested:

```
impl<'a, T, S> LockableScore<'a> for S
	where S: Deref<Target = T>,
		T: LockableScore<'a>,
		T::Locked: Writeable {
	type Locked = T::Locked;

	fn lock(&'a self) -> Self::Locked {
		self.deref().lock()
	}
}
```

but it looks like it may be too generic as it conflicts with our implementation of `LockableScore<'a>` for `Mutex` and `RefCell` a couple of lines below. thoughts?

Reply:
Ah, yea, lets not go that way - instead lets just make the `WriteableScore` we pass to the `BackgroundProcessor` actually a `Deref<WriteableScore>`, then we can drop both of the Arc implementations. You'll need to have two bounds - one for the `WriteableScore` and one for the `Deref` as you'll want to re-use the `WriteableScore` bound on the `Persister.